charts/tekton-pipelines/templates/500-triggerTemplate.yaml

```diff
@@ -10,10 +10,10 @@ metadata:
     release: {{ $.Release.Name }}
     heritage: {{ $.Release.Service }}
 spec:
-  {{- if .params -}}
-  {{- if gt (len .params) 0 }}
+  {{- if .pipeline_params -}}
```

Comment:
I suspect this needs to exist if either params or pipeline params exists and loop over both

Reply:
I reread https://tekton.dev/docs/pipelines/pipelines/#specifying-parameters

> You can specify global parameters, such as compilation flags or artifact names, that you want to supply to the Pipeline at execution time. Parameters are passed to the Pipeline from its corresponding PipelineRun and can replace template values specified within each Task in the Pipeline.

Here is another source of inspiration: https://access.redhat.com/documentation/en-us/openshift_container_platform/4.5/html-single/pipelines/index#about-pipelines_understanding-openshift-pipelines

Check the title for item (6):

> Parameters used across all the Tasks in the Pipeline.

and here is my interpretation (let me know if you agree/disagree) - "global pipeline parameters" (i.e. pipeline_params), which in Tekton are listed above tasks, and task parameters unfortunately are labeled as "params" in Tekton. My impression is that when creating the PipelineRun we only supply values for the "global pipeline parameters". I have almost convinced myself of this. If it was not the design, it still makes sense to separate task parameters from pipeline parameters for these reasons: 1) have a single location in the script where parameters are set 2) looping over both task and pipeline parameters might result in assigning values to the same parameter more than once